torch/optim/adam.py

```diff
@@ -37,7 +38,7 @@ def __init__(self,
             raise ValueError(f"Invalid beta parameter at index 1: {betas[1]}")
         if not 0.0 <= weight_decay:
             raise ValueError(f"Invalid weight_decay value: {weight_decay}")
-
+        self._decoupled_weight_decay = decoupled_weight_decay
```

Comment:
We should store this as a part of the group, similar to capturable. For an example, please look at radam.py or nadam.py. 

There are also code changes needed in __set_state__

Reply:
You should also add a serialization test in test_optim.py that made sure that we call the right code path when deserializing a saved AdamW model (that decoupled_weight_decay is True).

Could you also locally verify that after this change, Adam and AdamW compute different things? Just to make sure the wiring is all correct.

Reply:
Done! Modeled the serialization test based on one I saw in test_nestedtensor.py, please let me know if there's a better way I should implement it :)